packages/CodingStyle/src/Rector/FuncCall/FunctionCallToConstantRector.php

```diff
@@ -0,0 +1,191 @@
+use PhpParser\Node\Expr\BinaryOp\SmallerOrEqual;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Scalar\String_;
+use Rector\Exception\ShouldNotHappenException;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\FuncCall\FunctionCallToConstantRector\FunctionCallToConstantRectorTest
+ */
+final class FunctionCallToConstantRector extends AbstractRector
+{
+    /**
+     * @var string[][]
+     */
+    private $operatorToComparison = [
+        '=' => Identical::class,
+        '==' => Identical::class,
+        'eq' => Identical::class,
+        '!=' => NotIdentical::class,
+        '<>' => NotIdentical::class,
+        'ne' => NotIdentical::class,
+        '>' => Greater::class,
+        'gt' => Greater::class,
+        '<' => Smaller::class,
+        'lt' => Smaller::class,
+        '>=' => GreaterOrEqual::class,
+        'ge' => GreaterOrEqual::class,
+        '<=' => SmallerOrEqual::class,
+        'le' => SmallerOrEqual::class,
+    ];
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Changes use of function calls to use constants', [
+            new CodeSample(
+                <<<EOS
+class SomeClass
+{
+    public function run()
+    {
+        version_compare(PHP_VERSION, '5.3.0', '<');
+    }
+}
+EOS
+                ,
+                <<<EOS
+class SomeClass
+{
+    public function run()
+    {
+        PHP_VERSION_ID < 50300;
+    }
+}
+EOS
+            ),
+            new CodeSample(
+                <<<EOS
+class SomeClass
+{
+    public function run()
+    {
+        $value = php_sapi_name();
+    }
+}
+EOS
+                ,
+                <<<EOS
+class SomeClass
+{
+    public function run()
+    {
+        $value = PHP_SAPI;
+    }
+}
+EOS
+            ),
+            new CodeSample(
+                <<<EOS
+class SomeClass
+{
+    public function run()
+    {
+        $value = pi();
+    }
+}
+EOS
+                ,
+                <<<EOS
+class SomeClass
+{
+    public function run()
+    {
+        $value = M_PI;
+    }
+}
+EOS
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isName($node, 'version_compare')) {
+            return $this->refactorVersionCompare($node);
+        }
+
+        if ($this->isName($node, 'php_sapi_name')) {
+            return new ConstFetch(new Name('PHP_SAPI'));
+        }
+
+        if ($this->isName($node, 'pi')) {
+            return new ConstFetch(new Name('M_PI'));
+        }
+
+        return null;
+    }
+
+    private function refactorVersionCompare(Node $node): ?Node
```

Comment:
```suggestion
    private function refactorVersionCompare(FuncCall $funcCall): ?BinaryOp
```

Here you can use more precise types